components/kubeflow-pipeline/pipeline_components/analytics-client/flwr_analytics_client/fedhist.py

```diff
@@ -0,0 +1,50 @@
+from typing import Tuple
+
+import data
+import numpy as np
+from diffprivlib.tools import histogram as histogram_dp
+from flwr.common import Config, Properties
+from provider import AnalyticsProvider, numpy_to_scalar
+
+
+class HistogramProvider(AnalyticsProvider):
+    @property
+    def name(self) -> str:
+        return "histogram"
+
+    def get_properties(self, config: Config) -> Properties:
+        d = data.data()
+
+        nbins = config["nbins"]
+        hrange = (config["hmin"], config["hmax"])
+
+        counts, bins = compute_histogram(d, nbins=nbins, hrange=hrange)
+
+        return {
+            "counts": numpy_to_scalar(counts),
+            "bins": numpy_to_scalar(bins),
+        }
+
+
+def compute_histogram(
+    input: np.ndarray,
+    nbins: int,
+    hrange: Tuple[float, float],
+    diffpriv: bool = False,
+    epsilon: float = 1.0,
+) -> Tuple[np.ndarray, np.ndarray]:
+    if diffpriv is True:
+        counts, bins = histogram_dp(
+            input,
+            bins=nbins,
+            range=hrange,
+            epsilon=epsilon,
+        )
+    else:
+        counts, bins = np.histogram(
+            input,
+            bins=nbins,
+            range=hrange,
+        )
+
+    return counts, bins
```

Comment:
If the `compute_histogram` function is only relevant for the `HistogramProvider` I would propse to move this inside the `get_properties` method:

```suggestion
        counts, bins = np.histogram(
            input,
            bins=nbins,
            range=hrange,
        )
        
        return {
            "counts": numpy_to_scalar(counts),
            "bins": numpy_to_scalar(bins),
        }
```

Notice that I've also removed the `if diffpriv is True` because the `get_properties` method was not making use of it.

Reply:
This is to have a clear distinction between the provider code and the actual algorithm. Though in this particular case the algorithm itself is very simple, IMO it's still beneficial to have this distinction.

Reply:
Could you elaborate on the benefit of having the distinction?. So far the only benefit I see is to be able to unit-test the algorithm, but that can also be achieved through the provider hence my suggestion of putting inside the provider.

Reply:
It could be helpful once these algorithms get more complicated. But it isn't really helping in this particular case. I'll add it to `get_properties`.